activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-hal-rest-api/src/test/java/org/activiti/rest/SimpleRestTemplateBuilder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.rest;
+
+import java.util.Collections;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.hateoas.hal.Jackson2HalModule;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SimpleRestTemplateBuilder {
+
+    public TestRestTemplate build(String rootUri) {
```

Comment:
Instead of creating a type here (a new class) I would rather do a @Bean injection like the example that I share previously. So you can auto wire the TestRestTemplate. We need to make sure that the RestTemplate itself is auto wired, because as we find out there is some extra magic injected to the object and the request headers for integrating with other frameworks.